os/arch/arm/src/imxrt/imxrt_wdog_lowerhalf.c

```diff
@@ -0,0 +1,256 @@
+	DEBUGASSERT(priv);
+
+	imxrt_wdog_init(priv->base, &priv->config);
+	return OK;
+}
+
+static int imxrt_wdog_stop(FAR struct watchdog_lowerhalf_s *lower)
+{
+	struct imxrt_wdog_lowerhalf_s *priv = (struct imxrt_wdog_lowerhalf_s *)lower;
+	DEBUGASSERT(priv);
+
+	imxrt_wdog_deinit(priv->base);
+
+	return OK;
+}
+
+static int imxrt_wdog_keepalive(FAR struct watchdog_lowerhalf_s *lower)
+{
+	struct imxrt_wdog_lowerhalf_s *priv = (struct imxrt_wdog_lowerhalf_s *)lower;
+	DEBUGASSERT(priv);
+
+	imxrt_wdog_refresh(priv->base);
+
+	return OK;
+}
+
+static int imxrt_wdog_getstatus(FAR struct watchdog_lowerhalf_s *lower, FAR struct watchdog_status_s *status)
+{
+	struct imxrt_wdog_lowerhalf_s *priv = (struct imxrt_wdog_lowerhalf_s *)lower;
+	DEBUGASSERT(priv);
+
+	status->flags = imxrt_wdog_getstatusflags(priv->base);
+	status->timeout = priv->config.timeoutValue;
+	/* TO BE UPDATED */
+	status->timeleft = 0;
+
+	return OK;
+}
+
+static int imxrt_wdog_settimeout(FAR struct watchdog_lowerhalf_s *lower, uint32_t timeout)
+{
+	struct imxrt_wdog_lowerhalf_s *priv = (struct imxrt_wdog_lowerhalf_s *)lower;
+	DEBUGASSERT(priv);
+
+	/* Set to new timeout. */
+	priv->config.timeoutValue = (uint16_t)timeout;
+
+	/* Restart the watchdog with new configuration. */
+	imxrt_wdog_init(priv->base, &priv->config);
+
+	return OK;
+}
+
+static xcpt_t imxrt_wdog_capture(FAR struct watchdog_lowerhalf_s *lower, xcpt_t handler)
+{
+	struct imxrt_wdog_lowerhalf_s *priv = (struct imxrt_wdog_lowerhalf_s *)lower;
+	xcpt_t oldhandler;
+
+	DEBUGASSERT(priv);
+
+	oldhandler = priv->handler;
+
+	/* Set the new handler. */
+	priv->handler = handler;
+
+	/* Restart the watchdog with new configuration. */
+	imxrt_wdog_init(priv->base, &priv->config);
+
+	return oldhandler;
+}
+
+static int imxrt_wdog_ioctl(FAR struct watchdog_lowerhalf_s *lower, int cmd, unsigned long arg)
+{
+	struct imxrt_wdog_lowerhalf_s *priv = (struct imxrt_wdog_lowerhalf_s *)lower;
+	DEBUGASSERT(priv);
+
+	switch (cmd) {
+	case WDIOC_PAUSE: {
+		/* Set WDOG_WCR_WDW to 1 for suspending WDOG timer operation. */
+		priv->config.workMode.enableWait = 1;
+
+		/* Restart the watchdog with new configuration. */
+		imxrt_wdog_init(priv->base, &priv->config);
+	}
+	break;
+
+	case WDIOC_RESUME: {
+		/* Reset WDOG_WCR_WDW to continue WDOG timer operation. */
+		priv->config.workMode.enableWait = 0;
+
+		/* Restart the watchdog with new configuration. */
+		imxrt_wdog_init(priv->base, &priv->config);
+	}
+	break;
+
+	default: {
+		llvdbg("invalid cmd for wdog ioctl.\n");
+		return ERROR;
+	}
+	}
+	return OK;
+}
+
+/****************************************************************************
+ * Name: imxrt_wdog_initialize
+ *
+ * Description:
+ *   Initialize and register the imxrt watchdog. Watchdog timer is registered
+ *   as 'devpath'.
+ *
+ ****************************************************************************/
+int imxrt_wdog_initialize(const char *devpath, int wdog)
```

Comment:
renameing  wdog to something more practically(wdog_type or type or something...) will be nice..